imblearn/under_sampling/edited_nearest_neighbours.py

```diff
@@ -382,3 +382,171 @@ def _sample(self, X, y):
             return X_resampled, y_resampled, idx_under
         else:
             return X_resampled, y_resampled
+
+
+class AllKNN(SamplerMixin):
+    """Class to perform under-sampling based on the AllKNN method.
+
+    Parameters
+    ----------
+    return_indices : bool, optional (default=False)
+        Whether or not to return the indices of the samples randomly
+        selected from the majority class.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by np.random.
+
+    size_ngh : int, optional (default=3)
+        Size of the neighbourhood to consider to compute the average
+        distance to the minority point samples.
+
+    kind_sel : str, optional (default='all')
+        Strategy to use in order to exclude samples.
+
+        - If 'all', all neighbours will have to agree with the samples of
+        interest to not be excluded.
+        - If 'mode', the majority vote of the neighbours will be used in
+        order to exclude a sample.
+
+    n_jobs : int, optional (default=-1)
+        The number of thread to open when it is possible.
+
+    Attributes
+    ----------
+    min_c_ : str or int
+        The identifier of the minority class.
+
+    max_c_ : str or int
+        The identifier of the majority class.
+
+    stats_c_ : dict of str/int : int
+        A dictionary in which the number of occurences of each class is
+        reported.
+
+    X_shape_ : tuple of int
+        Shape of the data `X` during fitting.
+
+    Notes
+    -----
+    The method is based on [1]_.
+
+    This class supports multi-class.
+
+    Examples
+    --------
+
+    >>> from collections import Counter
+    >>> from sklearn.datasets import fetch_mldata
+    >>> from imblearn.under_sampling import AllKNN
+    >>> pima = fetch_mldata('diabetes_scale')
```

Comment:
Can you change the example to avoid fetching data. It can be a problem at testing.